src/client.js

```diff
@@ -1167,19 +1435,27 @@ MatrixClient.prototype.createKeyBackupVersion = function(info) {
         algorithm: info.algorithm,
         auth_data: info.auth_data,
     };
-    return this._crypto._signObject(data.auth_data).then(() => {
-        return this._http.authedRequest(
-            undefined, "POST", "/room_keys/version", undefined, data,
-            {prefix: httpApi.PREFIX_UNSTABLE},
-        );
-    }).then((res) => {
-        this.enableKeyBackup({
-            algorithm: info.algorithm,
-            auth_data: info.auth_data,
-            version: res.version,
-        });
-        return res;
+
+    // Now sign the backup auth data. Do it as this device first because crypto._signObject
+    // is dumb and bluntly replaces the whole signatures block...
+    // this can probably go away very soon in favour of just signing with the SSK.
+    await this._crypto._signObject(data.auth_data);
+
+    if (this._crypto._crossSigningInfo.getId()) {
```

Comment:
I am probably just forgetting something obvious at the moment, but why are we signing the key backup with the cross-signing master key? The key backup key isn't going away, so what's the purpose of this?

Reply:
This is so that device can know that the key backup is trusted without having get the private part of the key backup key and compare it's public part against the one on the account key backup.

Reply:
Hmm, okay... Is this described in the MSCs somewhere, or is it something we've crafted for Riot? At the moment, it feels like there are various details around how SSSS and cross-signing things weave together with key backup that don't seem to be recorded or discussed as an MSC yet, but may need to have one...?

Reply:
It's in the key backup MSC. https://github.com/matrix-org/matrix-doc/pull/1538/files#diff-1752bd09cd2950ace54b7efcd4e5ce87R495-R497

Reply:
Aha, so it is! 😁 I guess that's sufficient then.